modules/nixos/common/update.bash

```diff
@@ -0,0 +1,33 @@
+arch=$(uname -m)
+hostname=$(uname -n)
+p=$(curl -L https://buildbot.nix-community.org/nix-outputs/nix-community/infra/master/"$arch"-linux.host-"$hostname")
+
+if [[ "$(readlink /run/booted-system)" == "$p" ]]; then
+  return
+fi
+if [[ "$(readlink /run/current-system)" == "$p" ]]; then
+  return
+fi
+
+nix-store --realise "$p"
+nix-env --profile /nix/var/nix/profiles/system --set "$p"
+
+booted="$(readlink /run/booted-system/{initrd,kernel,kernel-modules} && cat /run/booted-system/kernel-params)"
+built="$(readlink "$p"/{initrd,kernel,kernel-modules} && cat "$p"/kernel-params)"
+if [[ $booted != "$built" ]]; then
+  if [[ -e /run/current-system ]]; then
+    echo "--- diff to current-system"
+    nvd diff /run/current-system "$p"
+    echo "---"
+  fi
+  /nix/var/nix/profiles/system/bin/switch-to-configuration boot
+  # don't use kexec if system is virtualized, reboots are fast enough
+  if ! systemd-detect-virt -q; then
+    kexec --load "$p"/kernel --initrd="$p"/initrd --append="$(cat "$p"/kernel-params) init=$p/init"
+  fi
+  if [[ ! -e /run/systemd/shutdown/scheduled ]]; then
```

Comment:
Shouldn't it than also create this file?

Reply:
The file is created by systemd when a shutdown/reboot is scheduled.